annex/restore-test/Vagrantfile

```diff
@@ -0,0 +1,66 @@
+# -*- mode: ruby -*-
+
+# Older versions of vagrant can't start the ubuntu bentobox with
+# private networking
+Vagrant.require_version ">= 1.8.2"
+
+box = "bento/ubuntu-16.04"
+hostname = 'annex-restore-test'
+
+memory = 2048 # in MB
+disk_size = 60 # in GB
+
+ip = '192.168.80.10'
+domain = 'localdomain'
+
+# Pass through github token
+env = { 
+  "VAULT_AUTH_GITHUB_TOKEN" => ENV["VAULT_AUTH_GITHUB_TOKEN"],
+  "VAULT_ADDR" => ENV["VAULT_ADDR"]
+}
+
+Vagrant.configure(2) do |config|
+  # Basic properties
+  config.vm.box = box
+  config.vm.provider :virtualbox do |vbox|
+    vbox.gui = false
+    vbox.memory = memory
+  end
+  config.vm.hostname = "#{hostname}.#{domain}"
+  config.vm.network :private_network, ip: ip
+  config.vm.network "forwarded_port", guest: 15432, host: 15433
+  config.vm.synced_folder 'shared', '/vagrant'
+
+  # Configure a second disk
+  config.persistent_storage.enabled = true
+  config.persistent_storage.mountname = "data"
+  config.persistent_storage.filesystem = "ext4"
+  config.persistent_storage.mountpoint = "/mnt/data"
+  config.persistent_storage.size = disk_size * 1024
+  config.persistent_storage.location = "disk/#{hostname}.vdi"  
```

Comment:
This is just me copying from @richfitz's work on the CI system. I'm not sure this is right for two reasons:

1. Is persistent storage the right thing when this is a disposable VM? Is there some non-persistent way of getting a large disk?
2. I don't think my scripts actually use the /mnt/data partition, they just succeed at the moment because the annex backup isn't large.

So generally need some help from Rich here.

Reply:
1. I did not find an easy way of getting a large disk added other than this persistent approach.  I did wonder about modifying the plugin that we use to make it nonpersistent.  One option would be to, on provisioning, do a `rm -rf /mnt/data/<whatever directory you're using>` to clear it out (or a `mkfs.xxx` call)
2. These lines: https://github.com/vimc/montagu/blob/master/staging/provision/setup-docker.sh#L9-L12 are how I move docker onto the big disk for the montagu staging provisioning.  That's where most of the big data lies.  Once the restore gets >30GB it will be necessary to copy that approach for the duplicati backup location too (`/montagu` ?)

Reply:
I've created https://vimc.myjetbrains.com/youtrack/issue/VIMC-1330 to do this in the future.